Report improvements

jontxo opened the pull request:
Hello

I have modified your project including several modifications that I though that they would be very helpfull.

1.  I have included a new parameter "disableMetadataIfnotPresent", that if it has the value true removes the columns of the metadata from the  features overview report. 

2. I have included a new parameter "showExecutionTime", that if it has the value true it shows the execution time in the features overview page and in the feature page.

3. I have added a new parameter "expandStepsNotPassed", that if it has the value true it expands the steps of each scenario failed in the feeature detail page

4. I have included a new parameter "featuresFolder", that can be used to pass the cucumber feature files used in the execution of the scenarios to the report.
    With this parameter the scenarios outline are shown in a different way. Instead of creating one for each row of the scenario outline, an scenario with all the scenarios joined is created. This scenario has the table with 
    the examples of the scenarios and if one of the rows of the table is clicked you can access the information of the execution of this row of the scenario outline.
    In the tests of the report I have included an example in the feature Scenarios outline.
    
5. Finally I have included a new parameter "navigateToFeatureIfThereIsOnlyOne", that if the report is going to be opened in a browser it opens it and navigates to the feature page in case there is only one feature.

Please let me know what do you think of these changes.
Regards

wswebcreation commented:
Hi @jontxo

Tnx for the PR, looks like a major job you did there. I'll take a look at it this week. Now need to get some rest 😁

jontxo commented:
Hello

Thanks for taking some time to review what I sent you. I work as a QA doing
automated testing in web services with java and I love to do some
programming in my free time. What we use to review the local executions is
the report generated by cucumber that is not very nice, that's why I have
started to see how the report could be improved. The easiest way I have
found is to start with a javascript node plugin; and I have started with
yours because I think is the best from the ones available. With it I can
create an html report starting from the json file, and in my case from the
feature files. It has been an interesting way of learning a lot about html,
css and javascript.

About the points you have some doubts about,
- showing the execution time in the report: I agree with your points, my
intention with the time is to help to schedule executions  and to identify
the scenarios that are taking too much time in order to see if they have
any problem.

- the new way of displaying the scenario outlines only makes the results
look a bit more like the feature they are obtained from, and I think it
makes the representation of these scenarios a bit more clearer.

- the new layout: it is only a first version and it is far from perfect, I
initially started modifying it because the old layout didn't work with the
new way of displaying the scenarios outline; and I continued because I
wanted to define a bit more some parts like the table with the features.

- the feature of the parameter "navigateToFeatureIfThereIsOnlyOne" is
helpful if your debugging scenarios and you do a quick execution to see the
log. It only saves you a mouse click once in a while but it is nice to have.

I would also like to include more things like execution history ,more
charts with the tags and the steps and a way to execute the plugin in
jenkins to have the report created and uploaded after each execution.

I know I have modified a lot your project and that it is not easy to review
it. I haven't sent any of the previous versions of what I have been doing
because they where a lot worse coded, and I didn't have nothing finished.
The first thing I finished was to make the plugin work with the
cucumber-jvm reports and the second was to get scenarios outline with the
new look. After that I added the last features much more easily. And when I
got to the point where I have finished the features that I had initially
thought about I created the pull request.

To complete the feature to have execution history will take some time, I'm
not even sure if doing it with javascript is the best option or if it would
be easier to have everything in java. Nevertheless I will let you know if I
get any results with it.

Regards

On 19 November 2017 at 18:59, Wim Selles <notifications@github.com> wrote:

> Hi @jontxo <https://github.com/jontxo>
>
> I've taken a quick look into it (couldn't let it go ;-)). First of all a
> nice job you did there. There are are a few points I really like and a few
> points I've got my doubts about.
> Let me start by saying I'm really enthusiastic about the way you picked
> this up.There are some nice additions (features and codewise) I really
> like, but there are also some things I don't like so I don't want to
> discourage you with the feedback below.
>
> The points I really like are:
>
> I have included a new parameter "disableMetadataIfnotPresent", that if it
> has the value true removes the columns of the metadata from the features
> overview report.
>
> Nice addition, I do think this shouldn't be a parameter, but should be
> default.
>
> I have added a new parameter "expandStepsNotPassed", that if it has the
> value true it expands the steps of each scenario failed in the feeature
> detail page
>
> Nice addition, I do think this shouldn't be a parameter, but should be
> default.
>
> The points I've got my doubts with:
>
> I have included a new parameter "showExecutionTime", that if it has the
> value true it shows the execution time in the features overview page and in
> the feature page.
>
> Personally I don't like to show this because in my opinion it's not
> representing the real situation / time. It's also always misinterpreted (at
> least what I've seen in projects). Times are many times counted to a total
> time but it could be that it was a parallel execution, it doesn't represent
> the total time, only the step execution time (so no spinning up the
> browser). The time doesn't say anything about the actual stuff that has
> been done in a step and so on. I still prefer not showing times in the
> report.
>
> I have included a new parameter "featuresFolder", that can be used to pass
> the cucumber feature files used in the execution of the scenarios to the
> report.
>
> I also have doubts about the use / advantage of this feature. I don't see
> it, sorry
>
> Finally I have included a new parameter "navigateToFeatureIfThereIsOnly
> One"
>
> It's a nice feature, but I doubt if this will be used a lot. So in my
> opinion there is some code implemented that makes it "more complex" but
> doesn't have many advantages looking at the time it is used.
>
> The new layout
>
> The olds layout was very simple. The new layout gives me a little bit the
> shivers, I don't like it. Lines are to hard and so on, sorry.
>
> I think it's better to split some features into small PR's, that makes it
> also easier to review. I need to think the coming days about how I'm going
> further with this.
>
> But again, tnx for the hard work!!!
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/wswebcreation/multiple-cucumber-html-reporter/pull/15#issuecomment-345536360>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AQuPSXiLYSHU0nR_NZ9N6Butjo7XSvqOks5s4GyJgaJpZM4QjUd9>
> .
>

wswebcreation commented:
I'm going to close this PR for now. Let's update the project in smaller steps like with the custom metadata.

I'm really thankful for all your work!!
